quill-async/src/main/scala/io/getquill/sources/async/AsyncSourceConfig.scala

```diff
@@ -14,8 +14,9 @@ import io.getquill.naming.NamingStrategy
 abstract class AsyncSourceConfig[D <: SqlIdiom, N <: NamingStrategy, C <: Connection](
   val name:          String,
   connectionFactory: Configuration => ObjectFactory[C]
-)
```

Comment:
**For the Postgres part:**
The Postgres JDBC driver adds RETURNING to the INSERT statement when I use `PreparedStatement prepareStatement(String sql, String columnNames[])` ([see](https://github.com/pgjdbc/pgjdbc/blob/f21f1689241920defe9c7beafde12d0b0f318ca0/pgjdbc/src/main/java/org/postgresql/jdbc/PgStatement.java#L1172)). Therefore, I had to make the postgres-async behave the same way. 

**For the MySQL part**
In order to return the generated value, I have to cast to MySQLQueryResult in order to get `lastInsertId'. 

My first implementation had everything coded inside `AsyncSource`, which was really ugly. So, I decided to create these two specialization so `AsyncSource` stays clean.